services/movie_session.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+from django.db.models.functions import TruncDate
+
+import init_django_orm  # noqa: F401
+from db.models import MovieSession, CinemaHall, Movie
+
+
+def create_movie_session(movie_show_time: str,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    movie = Movie.objects.get(id=movie_id)
+
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=cinema_hall,
+        movie=movie
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet | MovieSession:
+    queryset = MovieSession.objects.all()
+
+    if session_date:
+        date_object = datetime.strptime(session_date, "%Y-%m-%d").date()
+        queryset = queryset.annotate(
+            show_date=TruncDate("show_time")
+        ).filter(show_date=date_object)
+
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: str = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> None:
+    movie_session = MovieSession.objects.get(id=session_id)
```

Comment:
The `get` method will raise a `MovieSession.DoesNotExist` exception if no movie session with the given `id` is found. Consider handling this exception to avoid potential runtime errors.

Reply:
get_or_404

Reply:
+